src/xunit.v3.runner.common/Parsers/CommandLineParserBase.cs

```diff
@@ -513,6 +513,7 @@ void OnNoColor(KeyValuePair<string, string?> option)
 	{
 		GuardNoOptionValue(option);
 		Project.Configuration.NoColor = true;
+		Environment.SetEnvironmentVariable("NO_COLOR", "NO_COLOR");
```

Comment:
@bradwilson, I have an issue I am not sure how to solve.
when adding this line to `OnNoColor` the test below `SwitchDefault` fails with switch `-nocolor`.
after debugging I understood the issue. 
other tests that run before this test, call the `OnNoColor` method and then set the environment variable. and by the time we get to this test, the environment variable is already set, which messes up the test.
I made sure this was the problem by deleting the environment variable as seen below and then the test passes.

do you have any idea how I can solve this?

Reply:
Typically the approach you use when making a global change:

- Use [a custom test collection](https://xunit.net/docs/running-tests-in-parallel#parallelism-in-test-frameworks) for all the tests related to the global change, so that none of them run in parallel with one another;
- Ensure that the test cleans up (undoes) any global changes that it performed.

Taking a quick look, it might be best if the two `Switches` classes (one for in-proc console runner, one for out-of-proc console runner) were placed into a common custom test collection. Then, each could implement a constructor and `IDisposable.Dispose` to save the current value of `NO_COLOR` and restore it, respectively:

```csharp
[TestCollection("Switches Test Collection")]
public class Switches: IDisposable
{
    readonly string? originalNoColorValue;

    public Switches()
    {
        originalNoColorValue = Environment.GetEnvironmentVariable("NO_COLOR");
    }

    public void Dispose()
    {
        Environment.SetEnvironmentVariable("NO_COLOR", originalNoColorValue);
    }

    // ... existing tests ...
}
```

**_Also important here is that the test methods cannot be `static` any longer, since we do not create or dispose the test class for static tests._**

Reply:
added changes